torchok/constructor/constructor.py

```diff
@@ -281,22 +281,33 @@ def create_dataloaders(self, phase: Phase) -> List[DataLoader]:
             - ValueError: When OneOrOther composition is passed that isn't supported
         """
         if phase in self.hparams.data:
-            dataloaders = [
-                self._prepare_dataloader(phase_params.dataset, phase_params.dataloader)
-                for phase_params in self.hparams.data[phase] if phase_params is not None
-            ]
+            dataloaders = []
+            for phase_params in self.hparams.data[phase]:
+                if phase_params is not None:
+                    sampler_params = phase_params.get('sampler', None)
+                    dataloaders.append(self._prepare_dataloader(phase_params.dataset,
+                                                                phase_params.dataloader,
+                                                                sampler_params))
             dataloaders = dataloaders if len(dataloaders) > 1 else dataloaders[0]
             return dataloaders
         else:
             return []
 
     @staticmethod
-    def _prepare_dataloader(dataset_params: DictConfig, dataloader_params: DictConfig) -> DataLoader:
+    def _prepare_dataloader(dataset_params: DictConfig,
+                            dataloader_params: DictConfig,
+                            sampler_params: DictConfig) -> DataLoader:
         dataset = Constructor._create_dataset(dataset_params)
         collate_fn = dataset.collate_fn if hasattr(dataset, 'collate_fn') else None
-
+        if sampler_params is not None:
+            num_samples = sampler_params.params.get('num_samples', len(dataset))
```

Comment:
you need pop this because, if `sampler_params.params` has `num_samples` you put twice num_samples parameter to constructor
```
SAMPLERS.get(sampler_params.name)(num_samples=num_samples, **sampler_params.params)
```
But I use code for situations like this
```
if 'num_samples' not in sampler_params:
    sampler_params.params['num_samples'] = len(dataset)
```
In this case you not remove useful data from DictConfig. Suggest you to use this code.